### PR TITLE
Feature/bugfix/parser

### PR DESCRIPTION
**Pull request by alexjoybc**
# Description

This PR includes the following changes:

- Fix bug BCP-578, error when parsing the message
- Add documentation on how to contribute to the bc-paris repository

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Tested locally + unit test

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## For bcgov contributors:

this PR fixes jira ticket: **BCP-578**

### CODE REVIEW

File: src/test/java/ca/bc/gov/iamp/bcparis/message/MessageUtilsGetValueTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    public MessageUtilsGetValueTest(String input, String[][] expected) {
+        this.input = input;
+        this.expected = expected;
+    }
+
+    @Test
+    public void execute() {
+
+        for(int i = 0; i < expected.length; i++) {
```
Comment (adamkroonbcgov) [inline src/test/java/ca/bc/gov/iamp/bcparis/message/MessageUtilsGetValueTest.java]: is expected length every going to equal zero here?

File: src/main/java/ca/bc/gov/iamp/bcparis/message/MessageUtils.java [hunk 2]
```diff
@@ -33,21 +33,50 @@
...
+     *  <li>FLC
+     *  <li>VIN
+     *  <li>REG
+     *  <li>RNS
+     *  <li>RVL
+     *  <li>TestRNS
+     * </ul>
+     *
+     * @param message the source message
+     * @param key     a known key
```
Comment (adamkroonbcgov) [inline src/main/java/ca/bc/gov/iamp/bcparis/message/MessageUtils.java:62]: weird whitespace on this line?

File: src/main/java/ca/bc/gov/iamp/bcparis/message/MessageUtils.java [hunk 3]
```diff
@@ -34,23 +35,23 @@
...
         startIndex += key.length() + 1;
 
-        int currentEndIndex = message.length();
+        message = message.substring(startIndex);
 
-        for (String token : knownTokens) {
+        int currentEndIndex = message.length();
 
-            int tokenIndex = message.indexOf(token + SEMICOLLON);
+        for (String token : KNOWN_TOKENS) {
```
Comment (adamkroonbcgov) [inline src/main/java/ca/bc/gov/iamp/bcparis/message/MessageUtils.java:100]: I'm assuming this is like a javascript forEach, my java syntax knowledge is rusty. Will this ever reach an out of bounds?

File: src/test/java/ca/bc/gov/iamp/bcparis/message/MessageUtilsEdgeCaseGetValueTest.java [hunk 4]
```diff
@@ -26,4 +28,24 @@ public void WithNullStringShouldReturnNullString() {
         Assert.assertNull(result);
     }
 
+    @Test
+    public void Test() {
+        List<String> delimiters = Arrays.asList("SN:", "MT:", "MSID:", "FROM:", "TO:", "SUBJ:", "TEXT:", "RE:",
+                "TestRNS:", // Satellite test
```
Comment (adamkroonbcgov) [inline src/test/java/ca/bc/gov/iamp/bcparis/message/MessageUtilsEdgeCaseGetValueTest.java]: Can we remove inline comments, consolidate them into a one liner before the delimiters list? that way you've got your list as a one liner.

Comment (adamkroonbcgov) [reply src/test/java/ca/bc/gov/iamp/bcparis/message/MessageUtilsEdgeCaseGetValueTest.java]: Also not sure about best practices.

File: src/main/java/ca/bc/gov/iamp/bcparis/service/MessageService.java [hunk 5]
```diff
@@ -23,6 +24,13 @@
 			  "TEXT:${text}${re}" + NEW_LINE +
 			  NEW_LINE +
 			  "${icbc_response}";
+	private final String messageFormat = "SEND MT:M" + NEW_LINE +
+			"FMT:Y" + NEW_LINE +
```
Comment (adamkroonbcgov) [inline src/main/java/ca/bc/gov/iamp/bcparis/service/MessageService.java]: is this a one liner?

**Comment by adamkroonbcgov** [comment]
generally the commit messages for this monolith are descriptive except for commit 6adbce0. Not sure if you want to go back and do an amend to it. I'll address commit history and squashing like commits at a later date which will greatly help the review process for larger PRs but other than that it looks good given your review on teams as well.

File: src/main/java/ca/bc/gov/iamp/bcparis/message/MessageUtils.java (hunk 3 above)
Comment (alexjoybc) [reply src/main/java/ca/bc/gov/iamp/bcparis/message/MessageUtils.java:100]: yes it's a foreach loop

File: src/test/java/ca/bc/gov/iamp/bcparis/message/MessageUtilsEdgeCaseGetValueTest.java (hunk 4 above)
Comment (alexjoybc) [reply src/test/java/ca/bc/gov/iamp/bcparis/message/MessageUtilsEdgeCaseGetValueTest.java]: this is copy paste from https://github.com/bcgov/bcparis-service/blob/4e261104a342c6a2d545ba3b4b196fca06366252/src/main/java/ca/bc/gov/iamp/bcparis/transformation/MessageTransform.java#L60-L68

File: src/main/java/ca/bc/gov/iamp/bcparis/service/MessageService.java (hunk 5 above)
Comment (alexjoybc) [reply src/main/java/ca/bc/gov/iamp/bcparis/service/MessageService.java]: yes

**Review by alexjoybc — COMMENTED** [review]
this is really good feedback on the commit history. thanks @akroon3r

**Review by adamkroonbcgov — APPROVED** [review]

**Pull request merged**